### PR TITLE
Google Cloud Storage Bucket Destination

### PR DESCRIPTION
**Pull request by TamerShlash**
This pull request implements a google cloud storage destination. The destination requires the following initializer parameters:
- `client_email` and `private_key` for auth, just like the BigQuery steps.
- `bucket` the bucket name to which files will be uploaded.
- an `options` hash which can contain the following:
  - `base_local_path` this will be prepended the file paths when creating a UploadIO locally, but won't be added to the remote file name, defaults to nil (just use the given path).
  - `content_type` the content type of the file to be uploaded, defaults to `application/json`.
  - `upload_type` the google cloud [upload type](https://cloud.google.com/storage/docs/json_api/v1/how-tos/upload), defaults to `resumable`.

**Expected Input**: hash that contains a `:file` key, which will be used as the name of the file both locally (prepended by `local_base_path` if present) and remotely.
**Output**: hash identical to the input hash, but with the `file` key's value replaced with a CloudStorageObject object.

### CODE REVIEW

File: lib/optimus_prime/destinations/google_cloud_storage_bucket.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        @bucket = bucket
+        opts = default_options.merge(options)
+        @base_local_path = opts[:base_local_path]
+        @upload_type = opts[:upload_type]
+        @content_type = opts[:content_type]
+        @upload_jobs = []
+      end
+
+      def default_options
+        { base_local_path: nil, content_type: 'applicatoin/json', upload_type: 'resumable' }
```
Comment (Anumodana) [inline lib/optimus_prime/destinations/google_cloud_storage_bucket.rb]: Typo content type.

Comment (TamerShlash) [inline lib/optimus_prime/destinations/google_cloud_storage_bucket.rb]: As usual :panda_face:

**Comment by iammchat** [comment]
Look good for me :+1:

**Comment by Anumodana** [comment]
It looks good.

**Pull request merged**